pkg/imager/imager.go

```diff
@@ -342,3 +429,62 @@ func (i *Imager) buildUKI(ctx context.Context, report *reporter.Reporter) error
 
 	return nil
 }
+
+type overlayInstallerExecutor struct {
```

Comment:
let's move it out to a separate file, it could be an `overlay` subpackage in the `imager`, and add a type assertion on the interface to make it more clear what it does